src/frontend/components/sidebar.tsx

```diff
@@ -1,14 +1,55 @@
-import React from 'react'
-import MenuOptions from './menu-options'
+"use client"
+import React, { useEffect, useState } from 'react';
+import MenuOptions from './menu-options';
+import { IoIosArrowForward } from "react-icons/io";
 
-type Props = {}
+type Props = {
+  isOpen: boolean;
+  toggleSidebar: () => void;
+};
 
 const Sidebar = (props: Props) => {
-    return (
-        <aside className="sticky top-0 overflow-y-auto h-full bg-primary text-white py-1 px-2 w-64 md:block hidden">
-            <MenuOptions />
+  const [sidebarOpen, setSideBarOpen] = useState(true);
+
+  const handleViewSidebar = () => {
+    setSideBarOpen(!sidebarOpen);
+  };
+
+  useEffect(() => {
```

Comment:
You can utilize the `hooks/useWindowSize` hook to clean the code here.

Reply:
Done